### PR TITLE
Improve worker behavior in createFile

### PR DESCRIPTION
**Pull request by voddle**
### What changes are proposed in this pull request?

No more attempts to delete metadata when the path does not exist in createFile

### Why are the changes needed?

Improve worker behavior

### Does this PR introduce any user facing changes?

no

### CODE REVIEW

File: dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java [hunk 1]
```diff
@@ -808,13 +808,15 @@ public OpenFileHandle createFile(String path, CreateFilePOptions options)
...
+      if (exists) {
+        if (overWrite) {
+          // client is going to overwrite this file. We need to invalidate the cached meta and data.
+          mMetaManager.removeFromMetaStore(path);
+        } else {
+          throw new RuntimeException(
+              new FileAlreadyExistsException(
+                  String.format("File %s already exists but no overwrite flag", path)));
+        }
       }
```
Comment (YichuanSun) [inline dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java:820]: what if `exists == False` and `overWrite == True`? In the old version, the code will do `mMetaManager.removeFromMetaStore(path);`, while in your version, it does nothing here.

**Review by YichuanSun — CHANGES_REQUESTED** [review]
PTAL :)

File: dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java (hunk 1 above)
Comment (jiacheliu3) [reply dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java:820]: ^

Comment (voddle) [reply dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java:820]: i think we should do nothing in this case...? Or it is possible that a file could no exist in ufs but exist in MetaStore 🤔

Comment (huanghua78) [reply dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java:820]: UFS may change, but Alluxio does not sync with UFS.
So, it is possible that `exists == False` but metadata still has a metadata cache.
We should invalidate this metadata.

Comment (voddle) [reply dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java:820]: tyty

**Pull request closed without merging**